### PR TITLE
fix #273032: implement extensions to install soundfont, templates, etc.

### PR DESCRIPTION
**Pull request by anatoly-os**
Based on https://github.com/musescore/MuseScore/pull/3715

Added:
1. Progress bar when downloading packages
2. Message if extracting files from package is failed
3. Check packages updates on MuseScore startup; open Resource Manager if user wants to update packages
4. Refactoring in UpdateChecker, GUI changes in Resource Manager / Extensions tab

### CODE REVIEW

File: mscore/musescore.cpp [hunk 1]
```diff
@@ -2816,7 +2817,16 @@ bool MuseScore::hasToCheckForUpdate()
       if (ucheck)
             return ucheck->hasToCheck();
       else
-          return false;
+            return false;
+      }
+
+//---------------------------------------------------------
+//   hasToCheckForPackagesUpdate
```
Comment (lasconic) [inline mscore/musescore.cpp]: It's called Extension all over the place. Why package here ?

Comment (anatoly-os) [reply mscore/musescore.cpp]: Typo :)

**Comment by anatoly-os** [comment]
@lasconic I've replaced all my "packages" names with "extensions".

**Comment by lasconic** [comment]
See PR #3727

**Pull request closed without merging**